Make parameters optional in PATCH user_groups/<int:user_group_id>.

m-e-l-u-h-a-n opened the pull request:
View that handled `PATCH user_groups/<int:user_group_id>` required
  both name and description parameters to be passed. Due to this
  clients had to pass values for both these parameters even if
  one of them was changed.
  
  To resolve this name description parameters to
  `PATCH user_groups/<int:user_group_id>` are made optional.

Fixes: [CZO link](https://chat.zulip.org/#narrow/stream/378-api-design/topic/Required.20parameters.20for.20user.20group.20edit)

**Screenshots and screen captures:**

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented:
We need to document API changes in the API changelog. The new application feature tutorial and OpenAPI docs pages on ReadTheDocs cover how to do this, or just use `git log --follow -p api_docs/changelog.md` to get some examples.

m-e-l-u-h-a-n commented:
I wanted to ask if we need a mention about this in `changelog.md` as a Feature level entry?

timabbott commented:
Yes, any change to the actual API interface should be documented there. It may not be important, but it's less work to just consistently document them than be constantly making decisions for each API change about which are "important enough" to document. (It's only a tiny bit of work to add **Changes** entries etc., once one knows how)

m-e-l-u-h-a-n commented:
@timabbott I've updated the pull request to mention changes in `api_docs/changelog.md` and also added Changes entries for this endpoint in `zerver/openapi/zulip.yaml`.
This is ready for another review.

timabbott commented:
Merged, after some wording adjustments in the API docs, thanks @m-e-l-u-h-a-n!
